### PR TITLE
dem SUN

### PR DESCRIPTION
**Pull request by jbesteuos**

### CODE REVIEW

File: client/src/view/sun_view.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        let vbuf = VertexBuffer::new(context.get_facade(), &raw_vertex_buffer).unwrap();
+
+        let raw_index_buffer = [2, 1, 0, 3]; //TrianglesStrip
+
+        let ibuf = IndexBuffer::new(context.get_facade(),
+                                    PrimitiveType::TriangleStrip,
+                                    &raw_index_buffer)
+            .unwrap();
+
+        let prog = Program::from_source(context.get_facade(),
```
Comment (jonas-schievink) [inline client/src/view/sun_view.rs]: You can use `GameContext::load_program` instead (and move the shaders to the shader folder). This way you won't have to recompile plantex when changing a shader

File: client/src/view/sun_view.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                  &params)
+            .unwrap();
+    }
+
+    pub fn update(&mut self, pos: Vector3f) {
+
+        const SUN_SIZE: f32 = 35.0;
+        self.position = pos;
+        let raw_vertex_buffer = vec![
+            Vertex { i_position: [self.position.x+SUN_SIZE,
```
Comment (jonas-schievink) [inline client/src/view/sun_view.rs]: I don't think this is formatted right

File: client/src/view/sun_view.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        self.vertex_buffer = VertexBuffer::new(self.context.get_facade(), &raw_vertex_buffer)
+            .unwrap();
+
+    }
+}
+#[derive(Debug, Copy, Clone)]
+struct Vertex {
+    pub i_position: [f32; 3],
+    pub i_unit_coords: [f32; 3],
```
Comment (jonas-schievink) [inline client/src/view/sun_view.rs]: The third coordinate is always 0, could you use a `[f32; 2]` instead?

Comment (jbesteuos) [inline client/src/view/sun_view.rs]: No, third component is for the height

**Pull request merged**